### ISSUE TITLE
Spotify randomly closes on id5

### ISSUE THREAD

**Issue by Sticky70** [BimmerGestalt/AAIdrive#560]
Spotify through this app was great at the start but now randomly closes.
I've tried rebooting and no difference unfortunately.
Any suggestions please?

**Comment by pocketux** [comment]
Hi,

First of all thank you to bring us the opportunity to use this awesome app!

I'm having the same issue with my NBT Evo id5. Randomly I lost the control of the Spotify app (can't change the song, can't change the playlist, can't do anything). Other times I can't search in the library (all searchs returns empty results). And finally Spotify is closed in the car and I can't do anything to recover it, more than restart everything in my phone and sometimes even the NBT Evo.

It was working during some time but lately I'm having those issues. Could be any wrong config in my side? Or a conflict with any other apps using Bluetooth?

Thanks!

**Comment by pocketux** [comment]
Oh, I didn't know about the beta version. I've upgraded to the latest beta version and will be testing it during the next days.

**Comment by hufman** [comment]
Hmmm not sure what's going on! Sometimes Spotify just gets stuck and needs to be force-killed. AAIdrive hasn't had any recent updates about music control, so the beta probably won't help much.

**Comment by pocketux** [comment]
Don't know what's going on with my configuration but unfortunately right now it's totally unusable for me. Today I drive my car 4 for times for 6 to 8 kms each time and all the times I lost the control of Spotify app in the car and it was closed. 

I tried closing some other apps using Bluetooth and even turning off my Samsung smartwatch, but nothing changed. Tried also to force stop all the apps (Spotify, myBMW and AAIdrive) and also restart the smartphone (Oppo RenoZ), but no luck.

Don't know is I could test any other set up to get this working.

Thanks in advance!

**Comment by hufman** [comment]
I'm unfamiliar with this model of phone, but some phones are especially conservative with their power saving settings. Have you enabled background processing for Spotify? Android is supposed to keep the current music app running, but Spotify keeps asking me to enable background processing for some presumably justified reason!
Do any other music apps behave differently?
When you say "it was closed", does AAIdrive's car interface show the music app is <Not Connected> or does the entire AAIdrive app disconnect from the car?

**Comment by pocketux** [comment]
Yep, I enabled background processing for Spotify, myBMW and AAIdrive. Additionally I have disabled all the AI options of power saving on my phone since long time ago.

I didn't test any other music app. I'm going to test Amazon Music, or could I test any other?

Lately the entire AAIdrive app is disconnected from the car. The Spotify option disappears from the Idrive and I'm unable to recover it even disabling/enabling Bluetooth in the Idrive. When this happens, I change to Bluetooth Audio in Idrive and I'm able to listen the music from Spotify and even jump to the next or previous song, so Spotify is still alive in my phone. 

Thanks for your help!

**Comment by pocketux** [comment]
I have more information. I enabled Amazon Music, and had both available (Spotify and amazon Music). Tried to change between both several times and was working fine for some minutes but at some point I have lost the control of both apps in the Idrive, I was unable to open any of them, but music was still being played by Bluetooth audio. 

When that happened I opened the AAIdrive interface on my phone and I was able to see the Music Tab, and the Spotify control Window was there. I was able to jump between songs and even browse my library and change the play list, but nothing to do with the Idrive in my car. I had to force stop AAIdrive and open it again and then everything was restored in the car Idrive.

So, based on this, the issue does not look to be between AAIdrive and Spotify, but between AAIdrive and my car (F34 LCI with NBT Evo 8.8 ID5)

**Comment by pocketux** [comment]
I have fresh news on this. I guess I have found "the problem". Analyzing and testing to disable/uninstall other apps I have in my phone, above all, those that I recently installed, I have found which one was causing "the conflict".

I recently installed a 3rd party electronic immobilizer (igla 231) to my car, and I was using Author ID app to manage it. This app is connected by Bluetooth to the Igla ECU and is able to disable it automatically when I start your engine. It was working really fine for it purpose, but it looks to be  the guilty of the connection issue between AAIdrive and my NBT Evo.

I have removed it from my phone, and since that moment everything is again working as expected with AAIdrive. 

There is one single random issue still there, but it's a minor issue. Randomly, first time I click on Explore the search results are Empty and there is no way to get it working. I just disable/enable the Bluetooth in the Idrive and then it works fine. It just happened a couple of times, so I couldn't find any pattern yet. But definitely it's a minor issue.

Hope this can help others with the same issue.

Kind regards!
